src/mlpack/core/util/facilities.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file facilities.hpp
+ * @author Kirill Mishchenko
+ * @author Bisakh Mondal
+ *
+ * Utility for checking same size &  same dimensionality.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_UTIL_FACILITIES_HPP
+#define MLPACK_UTIL_FACILITIES_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace util {
+
+/**
+  * Check for if the given data points & labels have same size.
+  *
+  * @param data data.
+  * @param labels Labels.
+  * @param callerDescription A description of the caller that can be used for
+  *     error generation.
+  * @param mode For nature of comparision(default "CE").
+  *   types of mode:
+  *     "CE" equivalent to (data.n_cols, labels.n_elem).
+  *     "CC" equivalent to (data.n_cols, labels.n_cols).
```

Comment:
We can capture this automatically; we will always want to use `labels.n_elem` if `LabelsType` is `arma::Row<eT>` or `arma::Col<eT>`.  Do you think you can fix the code to do that?  Then we can remove the `mode` parameter. :+1:

Reply:
Thanks @rcurtin for pointing that out. I got biased by a quick glance at the MSE implementation & thought keeping this feature is necessary. 
https://github.com/mlpack/mlpack/blob/9672d5d15368aca5600e6f72cb8f2fd15d2bc25d/src/mlpack/core/cv/metrics/mse_impl.hpp#L18-L24
Yeah `n_elem` will be sufficient for this purpose.